### PR TITLE
Add direct dbt entrypoint

### PR DESCRIPTION
**Pull request by NiallRees**
Adds a `dbt` entrypoint which acts as a shortcut to the `dbtenv execute` command. Removes the need to add an alias to dbt. If we like this approach, resolves https://github.com/brooklyn-data/dbtenv/issues/19

The entrypoint is installed as a separate package, included in the extra `dbtenv[dbt-alias`.

### CODE REVIEW

**Comment by jaypeedevlin** [comment]
This is a good idea!  My only concern here is that any pre-existing dbt installs might have a fight for being the first in the path.  Not an issue for BDC users but potentially more widely.  Do you have any thoughts here?

**Review by danthelion — APPROVED** [review]
Overall I think it's a good proposition although I think this might conflict with pre-existing dbt installations / dbt development environments, etc. I would maybe consider creating a new alias which wraps `dbtenv execute --`.

**Comment by NiallRees** [comment]
> This is a good idea! My only concern here is that any pre-existing dbt installs might have a fight for being the first in the path. Not an issue for BDC users but potentially more widely. Do you have any thoughts here?

In my experimentation this actually makes it easier to work with other dbt installs (say in virtual environments). There's no alias any more, which means that as you say the first executable in the path wins. With the pipx example, if someone does:
1. `pipx install dbtenv`
2. then `dbt` will resolve to dbtenv
3. if they then install dbt in a virtual environment, then `dbt` will resolve to that activated virtual environment's dbt

However, attempting to install both dbtenv and dbt into the same environment does result in `dbt` only resolving to one of them (the latest one installed)

File: dbtenv/main.py [hunk 1]
```diff
@@ -126,7 +126,10 @@ def main(args: List[str] = None) -> None:
         uninstall_subcommand.add_args_parser(subparsers, [common_subcommand_args_parser])
 
         parsed_args = Args()
-        args_parser.parse_args(args, namespace=parsed_args)
+        if args_parser.prog == 'dbt':
+            # If the dbt entrypoint has been used, prefix the args with the execute subcommand
+            args = ['execute', '--'] + args
+            args_parser.parse_args(args, namespace=parsed_args)
```
Comment (NiallRees) [inline dbtenv/main.py:132]: @danthelion this is definitely nicer!

File: dbtenv/execute.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ def execute(self, args: Args) -> None:
                 break
 
         adapter_type = dbtenv.version.try_get_project_adapter_type(self.env.project_file, target_name=arg_target_name)
-        if args.dbt_version_specifier:
+        if getattr(args, 'dbt_version_specifier', None):
```
[the remark this answers is not in the document]
Comment (NiallRees) [reply dbtenv/execute.py]: Just updated with a much cleaner version

File: README.md [hunk 3]
```diff
@@ -49,37 +61,24 @@ dbtenv determines which dbt version to use by trying to read it from the followi
...
 - Run `dbtenv version --global <version>` to set the dbt version globally in the `~/.dbt/version` file. The `<version>` can be either a dbt version (e.g. 1.0.0) or full pip specifier (e.g. dbt-snowflake==1.0.0). dbtenv will attempt to automatically detect the required adapter or version from the environment if not specified.
 - Run `dbtenv version --local <version>` to set the dbt version for the current directory in a `.dbt_version` file. The `<version>` can be either a dbt version (e.g. 1.0.0) or full pip specifier (e.g. dbt-snowflake==1.0.0). dbtenv will attempt to automatically detect the required adapter or version from the environment if not specified.
 
 ### Running dbt through dbtenv
-Run `dbtenv execute -- <dbt arguments>` to execute the dbt version determined dynamically based on the current environment, or run `dbtenv execute --dbt <version> -- <dbt arguments>` to execute the specified dbt version.
+There are two ways of running dbt with dbtenv, the first is to use the `dbtenv execute` command, and the second is to install the `dbtenv-dbt-alias` package included in the `[dbt-alias]` extra, which created a `dbt` entrypoint which acts as a shortcut to `dbtenv execute`.
```
Comment (NiallRees) [inline README.md:70]:
```suggestion
There are two ways of running dbt with dbtenv, the first is to use the `dbtenv execute` command, and the second is to install the `dbtenv-dbt-alias` package included in the `[dbt-alias]` extra, creating a `dbt` entrypoint which acts as a shortcut to `dbtenv execute`.
```

**Comment by NiallRees** [comment]
@jaypeedevlin @danthelion I've separated out the `dbt` entrypoint installation into its own package. That means installing `dbtenv` as before now doesn't install the `dbt` entrypoint.

**Pull request merged**